types/setting.go

```diff
@@ -70,6 +70,7 @@ const (
 	SettingNameReplicaZoneSoftAntiAffinity                              = SettingName("replica-zone-soft-anti-affinity")
 	SettingNameNodeDownPodDeletionPolicy                                = SettingName("node-down-pod-deletion-policy")
 	SettingNameAllowNodeDrainWithLastHealthyReplica                     = SettingName("allow-node-drain-with-last-healthy-replica")
+	SettingNameNodeWithLastHealthyReplicaDrainPolicy                    = SettingName("node-with-last-healthy-replica-drain-policy")
```

Comment:
Is it possible to consider a more generic setting name, such as `node-drain-policy`? I am thinking if we can move `kubernetes-cluster-autoscaler-enabled` as a policy.

Reply:
> Is it possible to consider a more generic setting name, such as node-drain-policy

Great idea! Surely 

> I am thinking if we can move `kubernetes-cluster-autoscaler-enabled` as a policy.

How does this policy work? Is it correct that this policy would be equivalent to `block` because we need to protect last replica even if it is a stopped replica?

Reply:
> How does this policy work? Is it correct that this policy would be equivalent to block because we need to protect last replica even if it is a stopped replica?

Sorry I've confused you earlier.  Unfortunately, the logic for `kubernetes-cluster-autoscaler-enabled` doesn't work for this case because It will protect the last replica IM even if the volume is detached. Ref: https://github.com/longhorn/longhorn-manager/blob/master/controller/instance_manager_controller.go#L764-L780

Reply:
> 

Ok. I will create another policy `fall-back-to-allow-node-drain-with-last-healthy-replica-setting` as default policy for the new setting. In the future, we can drop support for that policy when the old setting is removed. What do you think?

Reply:
Thank you for the confirmation. Yes, I agree that `kubernetes-cluster-autoscaler-enabled` doesn't control whether we should remove the PDB for IM when the last replica of a volume is on the node. It is just a trigger condition for the function `canDeleteInstanceManagerPDB`. Another trigger is when a node becomes unscheduled.